coap-core/src/test/java/com/mbed/coap/transport/udp/DatagramSocketTransportTest.java

```diff
@@ -66,61 +65,48 @@ public void initializingWithStateException() throws IOException {
 
             trans.start(mock(CoapReceiver.class));
 
-            try {
-                trans.setReuseAddress(true);
-                fail();
-            } catch (Exception e) {
-                assertTrue(e instanceof IllegalStateException);
-            }
-            try {
-                trans.setSocketBufferSize(1234);
-                fail();
-            } catch (Exception e) {
-                assertTrue(e instanceof IllegalStateException);
-            }
         } finally {
             trans.stop();
         }
     }
 
     @Test
     public void initializeWithParameters() throws Exception {
-        DatagramSocketTransport trans = new DatagramSocketTransport(new InetSocketAddress(0), mock(Executor.class));
-        trans.setReuseAddress(false);
-        trans.setSocketBufferSize(12345);
+        DatagramSocketAdapter datagramSocket = new DatagramSocketAdapter(new InetSocketAddress(0), false, 12345);
+        DatagramSocketTransport trans = new DatagramSocketTransport(datagramSocket, mock(Executor.class));
         trans.start(mock(CoapReceiver.class));
 
-        assertTrue(trans.getSocket().isBound());
-        assertFalse(trans.getSocket().isClosed());
-        assertEquals(12345, trans.getSocket().getSendBufferSize());
-        assertFalse(trans.getSocket().getReuseAddress());
+        assertTrue(((DatagramSocketAdapter) trans.getSocket()).isBound());
```

Comment:
This is the same as `datagramSocket`, so why to cast?

Reply:
fixed.